edxapp/Dockerfile

```diff
@@ -23,15 +23,19 @@ VOLUME /openedx/data
 WORKDIR /openedx/edx-platform
 
 ## Checkout edx-platform code
-RUN git clone https://github.com/edx/edx-platform.git --branch open-release/ginkgo.master --depth 1 .
+ARG EDX_PLATFORM_REPOSITORY=https://github.com/edx/edx-platform.git
+ARG EDX_PLATFORM_VERSION=open-release/ginkgo.master
+RUN git clone $EDX_PLATFORM_REPOSITORY --branch $EDX_PLATFORM_VERSION --depth 1 .
+VOLUME /openedx/edx-platform
 
-# Install python requirements
-RUN pip install -r requirements/edx/pre.txt
-RUN pip install -r requirements/edx/github.txt
-RUN pip install -r requirements/edx/local.txt
-RUN pip install -r requirements/edx/base.txt
-RUN pip install -r requirements/edx/post.txt
-RUN pip install -r requirements/edx/paver.txt
+# Install python requirements (clone source repos in a separate dir, otherwise
+# will be overwritten when we mount edx-platform)
+RUN pip install --src ../venv/src -r requirements/edx/pre.txt
+RUN pip install --src ../venv/src -r requirements/edx/github.txt
+RUN pip install --src ../venv/src -r requirements/edx/local.txt
+RUN pip install --src ../venv/src -r requirements/edx/base.txt
+RUN pip install --src ../venv/src -r requirements/edx/post.txt
+RUN pip install --src ../venv/src -r requirements/edx/paver.txt
 
 # Install nodejs requirements
 RUN npm install
```

Comment:
node_modules is also overriden when you mount edx-platform.

Reply:
Hmmm I had not realised that... Is that an issue? Do we need node_modules at runtime?

Reply:
not at runtime but if you use this as a devstack and the developper is changing things on the frontend then he might need to recompile the js I presume?

Reply:
One solution could be to re-run "npm install" after mounting your local "edx-platform" ? I chose to "docker cp" the edx-platform from the container to my local disk before doing the mount but I'm not fully satisfied with the beauty of it :-/  We need a better idea...

Reply:
I'll assume that Open edX developers are aware that they need to recompile. So they will run `paver assets` in the devstack, wich will install node_modules in the mounted repository and will solve the problem.

I strongly believe that mounting edx-platform in a volume is the way to go: this is the only way to have seamless integration with quick reload in the devstack.